### ISSUE TITLE
representation of the producer in view metadata

### ISSUE THREAD

**Issue by keighrim** [clamsproject/mmif#153]
(also see #9)

---

We need to more specific representation of *producer* in a view metadata. Current specification ([0.2.2](https://mmif.clams.ai/0.2.2/schema/mmif.json)) only requires `app` field that an app can store its *name* as a string. In the past we discussed a necessity (or desire) of throwing more about the app (just [version](https://github.com/clamsproject/mmif/issues/9#issue-627427527), or [the entire appMetadata](https://github.com/clamsproject/mmif/issues/9#issuecomment-659617987) - the thing is the appMetadata is still underspecified (https://github.com/clamsproject/clams-apps/issues/3). )

**Comment by marcverhagen** [comment]
Yeah, just a string is not good enough. I was thinking this could be a URL on the App Directory site or for now the URL of the GitHub repository of the app.

**Comment by keighrim** [comment]
The commit above suggests an `app` object to have three children - `name`, `url`, and `parameters` 
* `name` is a simple string for app name 
* `url` is URI formatted string for app repository address or URI to app directory entry once directory is ready
* `parameter` is an object to store information regarding #154 .

**Comment by marcverhagen** [comment]
I guess the `name` property is technically not needed since it will show in the url. And I am not convinced we need an `app` object and introduce more structure to the view metadata.

**Comment by keighrim** [comment]
We can keep *app* field that's just holding URI of the app, and add *parameters* field as a sibling to *app* (and *contains*). I thought it's just a bit more sensible to have *parameters* as a child of *app* since parameters determine how the app works.

**Comment by marcverhagen** [comment]
```json
{
  "app": "http://appdirectory.clams.ai/tessearct/2.03",
  "parameters": {
    "use-boxes": "true" },
  "contains": { "TextDocument": {} }
}
```

**Comment by keighrim** [comment]
We wanted to keep the representation of the producer in the view metadata as simple as possible, so we decided to use a single URI-formatted string keyed by `app`. 
Although we haven't decided the URI assignment of apps (which will be done during finalizing the app directory design (https://github.com/clamsproject/clams-apps/issues/15)), we have a full control over how the URI string is formed. So I'm mentioning here for future reference that we want the URL to have at least app name and preferably app version.

**Comment by keighrim** [comment]
As we retired the usage of MMIF as the gold format in NER annotation dataset, I no longer see the necessity of a representation of human annotators in `view.metadata.app` (producer) field, in the foreseeable future. 

For machine-generated views, we all agree that the producer representation is simply the IRI format of the app identifier from [the appmetadata](https://sdk.clams.ai/appmetadata.html). 

Closing the issue as resolved.
